src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -5584,130 +5584,8 @@ CalcHeatBalanceInsideSurf( Optional_int_const ZoneToResimulate ) // if passed in
-						ShowContinueErrorTimeStamp( "" );
-						if ( ! zone.TempOutOfBoundsReported ) {
-							ShowContinueError( "Zone=\"" + zone.Name + "\", Diagnostic Details:" );
-							if ( zone.FloorArea > 0.0 ) {
-								ShowContinueError( "...Internal Heat Gain [" + RoundSigDigits( zone.InternalHeatGains / zone.FloorArea, 3 ) + "] W/m2" );
-							} else {
-								ShowContinueError( "...Internal Heat Gain (no floor) [" + RoundSigDigits( zone.InternalHeatGains / zone.FloorArea, 3 ) + "] W" );
-							}
-							if ( SimulateAirflowNetwork <= AirflowNetworkControlSimple ) {
-								ShowContinueError( "...Infiltration/Ventilation [" + RoundSigDigits( zone.NominalInfilVent, 3 ) + "] m3/s" );
-								ShowContinueError( "...Mixing/Cross Mixing [" + RoundSigDigits( zone.NominalMixing, 3 ) + "] m3/s" );
-							} else {
-								ShowContinueError( "...Airflow Network Simulation: Nominal Infiltration/Ventilation/Mixing not available." );
-							}
-							if ( zone.IsControlled ) {
-								ShowContinueError( "...Zone is part of HVAC controlled system." );
-							} else {
-								ShowContinueError( "...Zone is not part of HVAC controlled system." );
-							}
-							zone.TempOutOfBoundsReported = true;
-						}
-						ShowFatalError( "Program terminates due to preceding condition." );
-					} else {
-						ShowSevereError( "Temperature (high) out of bounds [" + RoundSigDigits( TH12, 2 ) + "] for zone=\"" + zone.Name + "\", for surface=\"" + surface.Name + "\"" );
-						ShowContinueErrorTimeStamp( "" );
-						if ( ! zone.TempOutOfBoundsReported ) {
-							ShowContinueError( "Zone=\"" + zone.Name + "\", Diagnostic Details:" );
-							if ( zone.FloorArea > 0.0 ) {
-								ShowContinueError( "...Internal Heat Gain [" + RoundSigDigits( zone.InternalHeatGains / zone.FloorArea, 3 ) + "] W/m2" );
-							} else {
-								ShowContinueError( "...Internal Heat Gain (no floor) [" + RoundSigDigits( zone.InternalHeatGains / zone.FloorArea, 3 ) + "] W" );
-							}
-							if ( SimulateAirflowNetwork <= AirflowNetworkControlSimple ) {
-								ShowContinueError( "...Infiltration/Ventilation [" + RoundSigDigits( zone.NominalInfilVent, 3 ) + "] m3/s" );
-								ShowContinueError( "...Mixing/Cross Mixing [" + RoundSigDigits( zone.NominalMixing, 3 ) + "] m3/s" );
-							} else {
-								ShowContinueError( "...Airflow Network Simulation: Nominal Infiltration/Ventilation/Mixing not available." );
-							}
-							if ( zone.IsControlled ) {
-								ShowContinueError( "...Zone is part of HVAC controlled system." );
-							} else {
-								ShowContinueError( "...Zone is not part of HVAC controlled system." );
-							}
-							zone.TempOutOfBoundsReported = true;
-						}
-						ShowFatalError( "Program terminates due to preceding condition." );
-					}
-				}
-			}
+			TestSurfTempCalcHeatBalanceInsideSurf( TH12, surface, zone, WarmupSurfTemp );
```

Comment:
So, maybe my suggestion to move this to a separate function to facilitate a unit test was not a good idea.  The only information that's required every time step is the current surface temperature.  The rest of this is only used to generate the error message if one of the limits is exceeded.  So maybe you should keep the outer if block here
```
if ( ( TH12 > MaxSurfaceTempLimit ) || ( TH12 < MinSurfaceTempLimit ) ) {
TestSurfTempCalcHeatBalanceInsideSurf( TH12, surface, zone, WarmupSurfTemp );
}
```